src/main/kotlin/Main.kt

```diff
@@ -96,6 +106,99 @@ private fun executeTests(tests: Collection<Test>): Iterable<ResultContext> {
     return fullResults
 }
 
+private fun generateTests(): Collection<Test> {
+    val allRelations = ProofSearcher().addAllProofs().findNewRelations(RelationLoader.relations)
+    //ProofLimiter(3).limit(allRelations)
+    return TestGenerator().addAllTests().generateTests(allRelations)
+}
+
+
+val OPERATORS = listOf("||", "\\\\", "&&", "consistency:", "refinement:")
+private fun sortTests(engine: EngineConfiguration, tests: Collection<Test>) : Collection<Test> {
+    var out = ArrayList(tests)
+
+    if (engine.queryComplexity != null) { //Query Complexity
+        val upper: Int; val lower: Int
+        if (engine.queryComplexity.size >= 2) {
+            upper = engine.queryComplexity[1]
+            lower = engine.queryComplexity.first()
+        } else {
+            upper = engine.queryComplexity.firstOrNull() ?: Int.MAX_VALUE
+            lower = 0
+        }
```

Comment:
If possible then I think engine should have a function which returns a Pair and then we could do something like this:
```
val (lower, upper) = engine.bounds()
```
and ``engine.bounds()`` could contain the marked logic